client/src/js/services/grid/TransactionService.js

```diff
@@ -532,6 +552,12 @@ function TransactionService($timeout, util, uiGridConstants, bhConstants, Notify
    * This function saves all transactions by
    */
   Transactions.prototype.save = function save() {
+    var clientErrors = validateTransaction(this._entity);
```

Comment:
The best way to do this would be to add a validation key at the top of this service, like:
```js
var ERR_IMBALANCED_TXN = 'POSTING_JOURNAL.ERRORS.UNBALANCED_TRANSACTIONS';

// later in validateTransaction()
var error;

if (debit !== credit) {
  error = ERR_UNBALANCED_TXN;
}

// another condition

// another condition ..

return error;
```